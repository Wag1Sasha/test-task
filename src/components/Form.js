import React, {useState} from 'react'
import { Button } from 'react-bootstrap';
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'
import {AppContainer} from '../styled-components/styledComponents'
import {ImgContainer} from '../styled-components/styledComponents'
// import {FooterWrapper} from '../styled-components/styledComponents'



export const Form =({sortLastTags, lastTags})=>{

  const [value, setValue] = useState('')
  const [data,setData] = useState([])
  


  const handleChange =(e)=>{
    setValue(e.target.value);
  }


  const handleSubmit =()=>{
    if (value.trim() === "") return;
    // sortLastTags()
    // lastTags.push(value)
    fetch(`https://pixabay.com/api/?key=19776687-7c6cc257a41e8830f998f821c&q=${value}`)
    .then(res =>res.json())
    .then(data=>setData(data.hits))
  }

  return (
    <>
      <AppContainer>
        <InputGroup className="mb-3">
        <FormControl
          type="text"
          placeholder="type tags..."
          aria-describedby="basic-addon2"
          value={value}
          onChange={handleChange}
          />
          <InputGroup.Append>
            <Button variant="outline-secondary" type="submit" onClick={handleSubmit}>Search</Button>
          </InputGroup.Append>
        </InputGroup>
      </AppContainer>
      {data.map((photo)=>(
        <ImgContainer key={Math.random()}>
          <img src={photo.largeImageURL} width="300px" height="300px" alt='search'></img>
        </ImgContainer>
      ))}
      {/* <FooterWrapper>
      {lastTags.map(tag=>(
        <h3 key={Math.random()}>{tag}</h3>
          ))}
      </FooterWrapper> */}
    </>
  );
}
