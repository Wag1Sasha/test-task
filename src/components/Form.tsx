import React from 'react';

import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Button from 'react-bootstrap/Button';

interface IFormProps {
  handleChange:(e: React.ChangeEvent<HTMLInputElement>)=>void,
  handleSubmit:(e: React.FormEvent<HTMLFormElement>)=>void,
  tags:ITagsProps
}


export const Form =({handleChange,handleSubmit,tags}:IFormProps)=>{

  return(
    <form onSubmit={handleSubmit}>
            <InputGroup className="mb-3">
              <FormControl
              type="text"
              placeholder="type tags..."
              aria-describedby="basic-addon2"
              name='tag1'
              value={tags.tag1 || ''}
              onChange={handleChange}
              />
              <FormControl
              type="text"
              name='tag2'
              placeholder="type tags..."
              aria-describedby="basic-addon2"
              value={tags.tag2 || ''}
              onChange={handleChange}
              />
              <FormControl
              type="text"
              name='tag3'
              placeholder="type tags..."
              aria-describedby="basic-addon2"
              value={tags.tag3 || ''}
              onChange={handleChange}
              />
              <InputGroup.Prepend>
                <Button variant="outline-secondary" type="submit">Search</Button>
              </InputGroup.Prepend>
            </InputGroup>
          </form>
  )
}