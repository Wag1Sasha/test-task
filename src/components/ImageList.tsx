import React from 'react';
import { ImgContainer, PhotoPageWrapper } from '../styled/styledPhotosPage';




export const ImageList = ({images}:IPropsImageList) => {


  return (
    <PhotoPageWrapper>
      {images.map((image,index) => (
        <ImgContainer key={index}>
          <img src={image.largeImageURL} width="300px" height="300px" alt='search'></img>
        </ImgContainer>
      ))}
    </PhotoPageWrapper>
  );
};