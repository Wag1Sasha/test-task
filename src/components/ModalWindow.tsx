import React from 'react';

import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';

export const ModalWindow = ({ image, show, handleClose }: IPropsModal) => (
  <Modal show={show} onHide={handleClose} centered size="xl">
    <img src={image} alt="bigPicture" />
    <Modal.Footer>
      <Button variant="secondary" onClick={handleClose}>
        Close
      </Button>
    </Modal.Footer>
  </Modal>
);
