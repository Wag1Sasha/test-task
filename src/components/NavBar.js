import React from 'react'
import Navbar from 'react-bootstrap/Navbar'
import {Nav} from 'react-bootstrap'

export const NavBar = ()=>{
  return(
    <>
  <Navbar bg="dark" variant="dark">
    <Navbar.Brand href="#home">Test-Task</Navbar.Brand>
    <Nav className="mr-auto">
      <Nav.Link href="/">Home</Nav.Link>
      <Nav.Link href="/photos">Search</Nav.Link>
    </Nav>
  </Navbar>
</>
  )
}