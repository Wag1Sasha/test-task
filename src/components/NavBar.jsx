import React from 'react'
import Navbar from 'react-bootstrap/Navbar';
import { Nav } from 'react-bootstrap';
import { Link } from 'react-router-dom';

export const NavBar = () => {
  return (
    <>
      <Navbar bg="dark" variant="dark">
        <Navbar.Brand href="#home">Test-Task</Navbar.Brand>
        <Nav className="mr-auto">
          <Link to="/">Home</Link>
        </Nav>
      </Navbar>
    </>
  );
};