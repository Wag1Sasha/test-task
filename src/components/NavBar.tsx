import React from 'react';

import { Link } from 'react-router-dom';
import Navbar from 'react-bootstrap/Navbar';
import { Nav } from 'react-bootstrap';

export const NavBar = () => (
  <Navbar bg="dark" variant="dark">
    <Navbar.Brand href="#home">Test-Task</Navbar.Brand>
    <Nav className="mr-auto">
      <Link to="/">Home</Link>
    </Nav>
  </Navbar>
);
