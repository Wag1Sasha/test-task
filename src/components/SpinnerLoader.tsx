import React from 'react';
import { Button, Spinner } from 'react-bootstrap';

export const SpinnerLoader = () => (
  <Button variant="dark" disabled>
    <Spinner as="span" animation="border" size="sm" role="status" aria-hidden="true" />
    Loading...
  </Button>
);
