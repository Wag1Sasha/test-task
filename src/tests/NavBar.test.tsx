import React from 'react';
import { shallow } from 'enzyme';

import { NavBar } from '../components/NavBar';

test('Render NavBar component', () => {
  const component = shallow(<NavBar />);
  expect(component).toMatchSnapshot();
});
