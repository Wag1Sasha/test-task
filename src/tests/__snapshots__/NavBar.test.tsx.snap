// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar component 1`] = `
<Fragment>
  <Navbar
    bg="dark"
    collapseOnSelect={false}
    expand={true}
    variant="dark"
  >
    <NavbarBrand
      href="#home"
    >
      Test-Task
    </NavbarBrand>
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <Link
        to="/"
      >
        Home
      </Link>
    </Nav>
  </Navbar>
</Fragment>
`;
